src/main/java/org/tron/common/runtime/vm/program/InternalTransaction.java

```diff
@@ -85,23 +85,31 @@ public InternalTransaction(Transaction trx,  InternalTransaction.TrxType trxType
     // outside transaction should not have deep, so use -1 to mark it is root.
     // It will not count in vm trace. But this deep will be shown in program result.
     this.deep = -1;
-    if (trxType.equals(TrxType.TRX_CONTRACT_CREATION_TYPE)) {
+    if (trxType == TrxType.TRX_CONTRACT_CREATION_TYPE) {
       CreateSmartContract contract = ContractCapsule.getSmartContractFromTransaction(trx);
       this.sendAddress = contract.getOwnerAddress().toByteArray();
       this.receiveAddress = EMPTY_BYTE_ARRAY;
       this.transferToAddress = Wallet.generateContractAddress(trx);
       this.note = "create";
       this.value = contract.getNewContract().getCallValue();
-    }
-    else if(trxType.equals(TrxType.TRX_CONTRACT_CALL_TYPE)){
+      this.data = contract.getNewContract().getBytecode().toByteArray();
+      if(this.data.length == 0){
+        this.data = null;
+      }
+    } else if(trxType == TrxType.TRX_CONTRACT_CALL_TYPE) {
       TriggerSmartContract contract = ContractCapsule.getTriggerContractFromTransaction(trx);
       this.sendAddress = contract.getOwnerAddress().toByteArray();
       this.receiveAddress = contract.getContractAddress().toByteArray();
       this.transferToAddress = this.receiveAddress.clone();
       this.note = "call";
       this.value = contract.getCallValue();
+      this.data = contract.getData().toByteArray();
+      if (this.data.length == 0) {
+        this.data = null;
+      }
+    } else {
+      // TODO: Should consider unknown type?
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1633&issues=AWak_C2fdPPRVdwyhXK4&open=AWak_C2fdPPRVdwyhXK4)

[ISSUE_KEY]: AWak_C2fdPPRVdwyhXK4